File: felicia/python/core/node/node_lifecycle.py
from felicia.python.core.node.node import _NodeLifecycle

class NodeLifecycle(_NodeLifecycle):
    def OnInit(self):
        print("OnInit called")
        self.on_init()

    def on_init(self):
        pass

    def OnDidCreate(self, node_info):
        print("OnDidCreate called")
        self.on_did_create(node_info)

    def on_did_create(self, node_info):
        pass

    def OnError(self, status):
        self.on_error(status)

    def on_error(self, status):
        pass

File: felicia/core/node/node_lifecycle.cc
// Copyright (c) 2019 The Felicia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "felicia/core/node/node_lifecycle.h"

namespace felicia {

NodeLifecycle::NodeLifecycle() = default;

NodeLifecycle::~NodeLifecycle() = default;

void NodeLifecycle::OnInit() {}

void NodeLifecycle::OnDidCreate(NodeInfo node_info) {}

void NodeLifecycle::OnError(Status status) { LOG(ERROR) << status; }

}  // namespace felicia